src/hooks/progress/use-collection-progress.ts

```diff
@@ -0,0 +1,44 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query'
+import useAuthentication from 'hooks/use-authentication'
+import {
+	getAllProgress,
+	GetAllProgressResult,
+} from 'lib/learn-client/api/progress'
+import { Collection } from 'lib/learn-client/types'
+
+type QueryDataType = GetAllProgressResult
+
+interface UseCollectionProgressOptions {
+	collectionId: Collection['id']
+}
+
+interface UseCollectionProgressResult
+	extends Omit<UseQueryResult<QueryDataType>, 'data'> {
+	data: UseQueryResult<QueryDataType>['data']
+}
+
+/**
+ * Handles checking if there is progress for the given `collectionId`.
+ */
+function useCollectionProgress({
```

Comment:
The `use-collection-progress` hook is necessary for tutorial completion toast display because we want to display a special "collection complete" toast when the last remaining incomplete tutorial in a collection is completed.